src/textual/widget.py

```diff
@@ -494,6 +494,16 @@ def window_region(self) -> Region:
         window_region = self.region.at_offset(self.scroll_offset)
         return window_region
 
+    @property
+    def region_with_margin(self) -> Region:
```

Comment:
Maybe this should be named `virtual_region_with_margin`. I've used the undecorated term "region" to mean the region relative to the screen.

Reply:
Updated